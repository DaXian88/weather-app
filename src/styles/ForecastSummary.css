.forecast-summary {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

div {
  margin: 5px;
}
