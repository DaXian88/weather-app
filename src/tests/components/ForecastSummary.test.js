import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 27081988,
    description: "stub description",
    icon: "stubIcon",
    temperature: { min: 12, max: 18 },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("renders the correct date, description, temperature and icon props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(getByText("27081988")).toHaveClass("forecast-summary_date");
    expect(getByText("stub description")).toHaveClass(
      "forecast-summary_description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary_icon");
    expect(getByText("18Â°C")).toHaveClass("forecast-summary_temperature");
  });
});
