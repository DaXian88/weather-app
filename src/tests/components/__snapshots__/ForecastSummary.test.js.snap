// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders the correct date, description, temperature and icon props 1`] = `[Function]`;
